sam-react-app/coreconcept/src/searchform/SearchForm.tsx

```diff
@@ -0,0 +1,32 @@
+import React, {Component, createElement, KeyboardEvent} from "react";
+
+class SearchForm extends Component {
+    state = {
+        keyword:"",
+    };
+    handleOnClick = () => {
+        this.searchOperation();
+    }
+    handleKeyDown = (event: KeyboardEvent<HTMLInputElement>) => {
+        if(event.key === "Enter"){
+            this.searchOperation();
+        }
+    }
+    searchOperation(){
+        const searchKeywordElement = document.getElementById("searchKeyword") as HTMLInputElement;
+        this.onSearch(searchKeywordElement.value);
+    }
+    onSearch(keyword: string){
+        console.log("searching for " + keyword);
+    }
+    render(){
+        return <div>
+            <p>-------------- Search form component --------------</p>
+            <input id='searchKeyword' onKeyDown={this.handleKeyDown} />
```

Comment:
You have to provide `value` to the input, that's how you sync input's value between updates.

Reply:
i use default value for initial value and there is an error saying I can either have defaultValue or value, but I cannot have them both.